src/client.test.ts

```diff
@@ -46,6 +47,30 @@ describe('Client', () => {
     expect(client.events.addEventListener).toHaveBeenCalledTimes(3)
   })
 
+  test('verify .onmessage() method behaviors', async () => {
+    const srcHost: string = 'src.example.com'
+    const tgtHost: string = 'tgt.example.com'
+    const tgtUrl: string = `http://${tgtHost}/web-hook`
+
+    const req = nock(`http://${tgtHost}`).post('/web-hook').query({ foo: 'bar', then: 'now' }).reply(200, 'Posted')
+
+    // We "receive" messages from src/channel5 and post to ${tgtUrl}.
+    const opts = { source: `http://${srcHost}/channel5`, target: tgtUrl }
+    const client = new Client(opts)
+
+    // Message that will be "received".
+    const msg = { data: `{"query": {"foo": "bar", "then": "now"}, "host": "${srcHost}", "body": "payload", "key1": "one", "key2": "two"}` }
+
+    // "Receive" the message.
+    client.onmessage(msg)
+
+    // Ensure all POST operations complete before test ends.
+    await flushPromises()
+    await flushPromises()
```

Comment:
We have to call it twice??

Reply:
Apparently.  I am not going to dig into this further.  This satisfied my curiosity about `flush-promises` and so I am considering this experiment "closed" as `gosmee` is our new "go-to" smee server/client.